From 83539bad34acdb50b3f454727098dcafd9360fef Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 12 Jul 2021 20:28:07 +0300
Subject: [PATCH] [SW-44616] gaudi2: update vuart registers

VUART registers are identified for gaudi2 from PCIE_ELBI block spare
registers, so update them.

ARM register offset: 0x4C14F60
ARC1 register offset: 0x4C15F60

Change-Id: Ic6ed0475be6b18c101a3e908cf74998640f1ebbc
---

diff --git a/arch/gaudi2/common.c b/arch/gaudi2/common.c
index 94f987e..119f823 100644
--- a/arch/gaudi2/common.c
+++ b/arch/gaudi2/common.c
@@ -385,8 +385,8 @@
 
 void preboot_delay(void) {}
 
-#define mmPCIE_WRAP_VUART_TX_DATA	0xC14F60
-#define mmPCIE_WRAP_VUART_TX_STATUS	0xC14F64
+#define mmPCIE_WRAP_VUART_TX_DATA	0x4C14F60
+#define mmPCIE_WRAP_VUART_TX_STATUS	0x4C14F64
 void preboot_putc_virt_uart(const char c)
 {
 	/*
diff --git a/arch/gaudi2_arc_gnu/common.c b/arch/gaudi2_arc_gnu/common.c
index 61cc714..8743f2c 100644
--- a/arch/gaudi2_arc_gnu/common.c
+++ b/arch/gaudi2_arc_gnu/common.c
@@ -47,9 +47,8 @@
 #define PCIE_LTR_DISABLE		0
 #define PCIE_CAP_HDR_NXT_OFFSET		0x1f8
 
-/* TODO: Correct these for Gaudi2 */
-#define mmPCIE_WRAP_VUART_TX_DATA       0xC15F60
-#define mmPCIE_WRAP_VUART_TX_STATUS     0xC15F64
+#define mmPCIE_WRAP_VUART_TX_DATA       0x4C15F60
+#define mmPCIE_WRAP_VUART_TX_STATUS     0x4C15F64
 #define HL_PCIE_LINK_RETRIES	50
 
 u32 get_efuse_read_reg(void)
@@ -252,11 +251,8 @@
 	 * DATA: mmPCIE_WRAP_VUART_TX_DATA
 	 * STATUS: mmPCIE_WRAP_VUART_TX_STATUS
 	 */
-	/* TODO: Enable later */
-	/*
-	 * regmap_writel(c, mmPCIE_WRAP_VUART_TX_DATA);
-	 * regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
-	 */
+	regmap_writel(c, mmPCIE_WRAP_VUART_TX_DATA);
+	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
 }
 
 unsigned int get_cs_pad_sel(void)
